### PR TITLE
#9944 Merge WebSocketCloseFrameHandler into WebSocketProtocolHandler …

### PR DESCRIPTION
**Pull request by doom369**
…in order to minimize pipeline

Motivation:

Handling of `WebSocketCloseFrame` is part of websocket protocol, so it's logical to put it within the `WebSocketProtocolHandler`. Also, removal of `WebSocketCloseFrameHandler` will decrease the channel pipeline.

Modification:

- `WebSocketCloseFrameHandler` code merged into `WebSocketProtocolHandler`. `WebSocketCloseFrameHandler` not added to the pipeline anymore
- Added additional constructor to `WebSocketProtocolHandler`
- `WebSocketProtocolHandler` now implements `ChannelOutboundHandler` and implements basic methods from it

Result:

`WebSocketCloseFrameHandler` is no longer used.

Fixes https://github.com/netty/netty/issues/9944

### CODE REVIEW

File: codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketCloseFrameHandler.java [hunk 1]
```diff
@@ -29,7 +29,9 @@
 
 /**
  * Send {@link CloseWebSocketFrame} message on channel close, if close frame was not sent before.
+ * @deprecated this class was merged into {@link WebSocketProtocolHandler} so should no longer be used
  */
+@Deprecated
 final class WebSocketCloseFrameHandler extends ChannelOutboundHandlerAdapter {
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketCloseFrameHandler.java]: @doom369 as this is package-private we can just remove the class imho.

Comment (doom369) [reply codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketCloseFrameHandler.java]: Removed

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@doom369 thanks a lot!

**Pull request merged**